core-rust/core-api-server/src/core_api/conversions/lts.rs

```diff
@@ -365,13 +371,99 @@ pub fn to_api_lts_fungible_resource_balance_change(
 }
 
 pub fn to_api_lts_resultant_account_fungible_balances(
-    _database: &StateManagerDatabase,
-    _context: &MappingContext,
-    _substate_changes: &BySubstate<ChangeAction>,
+    database: &StateManagerDatabase,
+    context: &MappingContext,
+    substate_changes: &BySubstate<ChangeAction>,
 ) -> Vec<models::LtsResultantAccountFungibleBalances> {
-    // TODO(parallel PR): go through substate changes, collect fungible vaults' new values, resolve
-    // each into its global component owning the vault, retain only accounts, adjust API docs.
-    vec![]
+    let fungible_vaults = substate_changes
+        .iter_node_ids()
+        .filter(|node_id| node_id.is_internal_fungible_vault())
+        .collect::<Vec<_>>();
+    let ancestries = database
+        .batch_get_ancestry(fungible_vaults.clone())
+        .into_iter()
+        .map(|record| record.expect("InternalFungibleVault does not have a parent"));
+    let node_id_to_ancestor: HashMap<_, _> = fungible_vaults.into_iter().zip(ancestries).collect();
+    let system_db_reader = SystemDatabaseReader::new(database);
+    let mut resultant_fungible_balances_by_account = HashMap::new();
+    substate_changes
+        .iter()
+        .filter_map(|(substate_reference, change_action)| {
+            let vault_id = &substate_reference.0;
+            if !vault_id.is_internal_fungible_vault() {
+                return None;
+            }
+
+            if substate_reference.1 != MAIN_BASE_PARTITION {
+                return None;
+            }
+
+            if substate_reference.2 != FungibleVaultField::Balance.into() {
+                return None;
+            }
+
+            let account_address = node_id_to_ancestor.get(vault_id).unwrap().parent.0;
+            let Ok(account_address) = GlobalAddress::try_from(account_address) else {
+                return None;
+            };
+
+            if !account_address.is_account() {
+                return None;
+            }
+
+            let vault_type_info = system_db_reader
+                .get_type_info(vault_id)
+                .expect("Vault missing TypeInfo substate");
+
+            let TypeInfoSubstate::Object(vault_type_info) = vault_type_info else {
+                panic!("TypeInfoSubstate of FungibleVault is not of type Object.");
+            };
+
+            assert!(
+                vault_type_info
+                    .blueprint_info
+                    .blueprint_id
+                    .eq(&BlueprintId::new(
+                        &RESOURCE_PACKAGE,
+                        FUNGIBLE_VAULT_BLUEPRINT,
+                    )),
+                "Vault's TypeInfo wrongly says this is not a fungible vault."
+            );
+
+            let resource_address =
+                ResourceAddress::new_or_panic(vault_type_info.get_outer_object().into());
+
+            let fungible_vault_balance_substate: FieldSubstate<VersionedFungibleVaultBalance> =
```

Comment:
This line works probably only thanks to incidental `sbor(transparent)`.
I recommend deserializing `FungibleVaultBalanceFieldSubstate` (which is `FieldSubstate<FungibleVaultBalanceFieldPayload>`) and then doing `.into_payload().into_latest()`

Reply:
Nice catch, thanks!